src/com/jdngray77/htmldesigner/backend/BackgroundTask.kt

```diff
@@ -1,25 +1,52 @@
 package com.jdngray77.htmldesigner.backend
 
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
 /**
  * Runs tasks in the background
  */
-object BackgroundTask {
+object BackgroundTask : Subscriber {
+
+    private val threadPool = Executors.newCachedThreadPool()
+
+    init {
+        EventNotifier.subscribe(this, EventType.USER_EXIT)
+    }
+
 
     fun runOnUIThread(runnable: Runnable) {
         TODO()
     }
 
+    
+    /**
+     * Submits a [runnable] task to the for [threadPool] for execution.
+     */
     fun submit(runnable: Runnable) {
-        TODO()
+        threadPool.submit(runnable)
     }
 
+
     /**
-     * - Disallow new tasks to be submitted,
-     * - wait here for all threads to close?
-     * - close the thread pool
+     * Shutdowns the application's [threadPool].
+     *
+     * Executes the [threadPool]'s remaining unfinished tasks and shuts it down on completion.
      */
-    fun shutdown() {
+    private fun shutdown() {
+        threadPool.shutdown()
 
+        try {
+            if (!threadPool.awaitTermination(10L, TimeUnit.SECONDS))
+                threadPool.shutdownNow() // terminates remaining tasks after 10 seconds
+        } catch (e: InterruptedException) {
+            return
+        }
+
+        threadPool.shutdownNow() // shutdown thread pool upon task(s) completion
     }
 
+    override fun notify(e: EventType) {
+        shutdown()
```

Comment:
**Crutial until the notification service is completed**

Add a guard clause to check that `e == EventType.USER_EXIT`.
The notification service is not complete, and just notifies on **every** event, so you have to check it.

Add a FIXME or TODO comment to the guard so we remember to remove it later.